src/main/java/com/google/googleidentity/oauth2/endpoint/AuthorizationEndpoint.java

```
@@ -0,0 +1,76 @@
+    You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package com.google.googleidentity.oauth2.endpoint;
+
+
+import com.google.googleidentity.oauth2.client.ClientSession;
+import com.google.googleidentity.security.UserSession;
+import com.google.googleidentity.user.UserDetails;
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+import freemarker.template.Configuration;
+import freemarker.template.TemplateException;
+import freemarker.template.Version;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.logging.Logger;
+
+/**
+ * Demo AuthorizationEndpoint for OAuth2 Server
+ *
+ */
+
+@Singleton
+public final class AuthorizationEndpoint extends HttpServlet {
+
+    private static final long serialVersionUID = 5L;
+
+    private static final Logger log = Logger.getLogger("AuthorizationCodeEndpoint");
+
+    @Inject
+    private final Provider<UserSession> userSession;
+
+    @Inject
+    private final Provider<ClientSession> clientSession;
+
+    @Inject
+    public AuthorizationEndpoint(Provider<UserSession> userSession, Provider<ClientSession> clientSession) {
+        this.userSession = userSession;
+        this.clientSession = clientSession;
+    }
+
+    public void init() throws ServletException {
+
+    }
+
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+
```

You can throw UnsupportedOperationException for now. Same for others.